core/src/main/scala/pickling/pickler/Array.scala

```diff
@@ -0,0 +1,12 @@
+package scala.pickling.pickler
+
+import scala.pickling._
+import scala.collection.generic.CanBuildFrom
+
+trait ArrayPicklers {
```

Comment:
All of these "ala carte" traits need to be documented in terms of what they can handle.

Additionally, it'd be ideal for those which could be statically generated and kept somewhere to be initiated in *one* place rather than regenerated via the implicit macros.   For array you should document that a compile-time-optimisation is to do that.